autoload/lsp/diag.vim

```diff
@@ -404,6 +404,9 @@ def DiagsUpdateLocList(bnr: number): bool
     setbufvar(bnr, 'LspQfId', getloclist(0, {id: 0}).id)
   endif
 
+  var LspQfNr: number = getloclist(0, {id: LspQfId, nr: 0}).nr
```

Comment:
is it correct? what if lspqfid is 0? or what if current is the right list?

Reply:
I'm also a bit skeptical for whether or not this will work for all cases 🤔

Reply:
Yeah, I believe it does work because `LspQfId` is either:

- The id of the existing error list if one previously existed
- 0 (equivalent to current list in `getloclist` `id` parameter) if this is a new error list

In the first case, it will make the existing list current. In the second case, it will make the new list the current list again since new lists are already made current when they are created. This is fine but unnecessary. I'll make it clearer.

Reply:
Cool, thanks for elaborating. What will happen if the history is exhausted by a lot of `:lgrep` etc commands by the user? AFAIK the history only contains the last 10 location lists.

Reply:
That case is handled here: https://github.com/yegappan/lsp/blob/ce8f74885cd32d60e761bb979966e2728bd8eb6c/autoload/lsp/diag.vim#L368-L371

Since the error list has fallen out of the stack, it is created anew, which automatically makes it the current error list.